CRM/Core/Permission/Joomla.php

```diff
@@ -131,4 +135,74 @@ public function checkGroupRole($array) {
     return FALSE;
   }
 
+  /**
+   * @inheritDoc
+   */
+  public function upgradePermissions($permissions) {
```

Comment:
... just think if you want to save some poor developer ten minutes of wondering or searching.